src/org/rascalmpl/util/maven/Artifact.java

```diff
@@ -151,6 +150,14 @@ private void calculateClassPath(Scope forScope, HashSet<WithoutVersion> alreadyI
         }
     }
 
+    private Artifact createSystemArtifact(Dependency d) {
+        var path = d.getSystemPath();
+        if (path == null) {
+            throw new RuntimeException("missing systemPath?");
```

Comment:
this is not in line with the other cases. If we cannot find a dependency, we report an error in the upper `messages` writer, and then return a placeholder artifact, so that down-stream users might choose to replace them with a local project reference.

And otherwise, not crash on this missing dependency, but just report it. 

Aside to this: this error message is non descriptive, and does not tell us what is missing, and if we would ran into this, we would need a breakpoint and a debugger to figure out why a user is having this exception.

Reply:
This was just defensive programming and not meant for user consumption. This check has now been removed in favor of propagating a null systemPath to a null path in the artifact.